webui/react/src/pages/ExperimentDetails/ExperimentVisualization.tsx

```diff
@@ -233,39 +234,42 @@ const ExperimentVisualization: React.FC<Props> = ({
     ).catch(() => {
       setPageError(PageError.MetricBatches);
     });
-
-    return () => canceler.abort();
-  }, [ activeMetric, experiment.id, filters.batch, isSupported ]);
+  }, [ activeMetric, experiment.id, filters.batch, isSupported, canceler.signal ]);
 
   // Set the default filter batch.
   useEffect(() => {
     if (!batches || batches.length === 0) return;
-    if (filters.batch !== DEFAULT_BATCH) return;
-    setFilters(prev => ({ ...prev, batch: batches.first() }));
-  }, [ batches, filters.batch ]);
+    setFilters(prev => {
+      if (prev.batch !== DEFAULT_BATCH) return prev;
+      return { ...prev, batch: batches.first() };
+    });
+  }, [ batches ]);
 
   // Validate active metric against metrics.
   useEffect(() => {
-    const activeMetricFound = (metrics || []).reduce((acc, metric) => {
-      return acc || (metric.type === activeMetric.type && metric.name === activeMetric.name);
-    }, false);
-    if (!activeMetricFound) setActiveMetric(searcherMetric.current);
-  }, [ activeMetric, metrics ]);
+    setActiveMetric(prev => {
+      const activeMetricFound = (metrics || []).reduce((acc, metric) => {
+        return acc || (metric.type === prev.type && metric.name === prev.name);
+      }, false);
+      if (!activeMetricFound) return (searcherMetric.current);
+      return searcherMetric.current;
```

Comment:
**question:** untested, but shouldn't this return `prev` if `activeMetricFound` is true?

Reply:
```
return activeMetricFound ? prev : searcherMetric.current;
```